test/Python/Poetry/CommonSpec.hs

```diff
@@ -0,0 +1,298 @@
+                          { gitUrl = "https://github.com/kennethreitz/requests.git"
+                          , gitRev = Nothing
+                          , gitTag = Nothing
+                          , gitBranch = Just "next"
+                          }
+                    )
+                  , ("my-packageUrl", PyProjectPoetryUrlDependencySpec $ PyProjectPoetryUrlDependency "https://example.com/my-package-0.1.0.tar.gz")
+                  , ("my-packageFile", PyProjectPoetryPathDependencySpec $ PyProjectPoetryPathDependency "../my-package/dist/my-package-0.1.0.tar.gz")
+                  , ("my-packageDir", PyProjectPoetryPathDependencySpec $ PyProjectPoetryPathDependency "../my-package/")
+                  , ("black", PyProjectPoetryDetailedVersionDependencySpec $ PyProjectPoetryDetailedVersionDependency "19.10b0")
+                  ]
+            , devDependencies =
+                Map.fromList
+                  [("pytest", PoetryTextVersion "*")]
+            }
+    }
+
+notSupportedPyProjectDependency :: PoetryDependency
+notSupportedPyProjectDependency = PyProjectPoetryPathDependencySpec $ PyProjectPoetryPathDependency{sourcePath = "../my-package/dist/my-package-0.1.0.tar.gz"}
+
+notSupportedPoetryLockDependency :: PoetryLockPackage
+notSupportedPoetryLockDependency =
+  PoetryLockPackage
+    { poetryLockPackageName = PackageName "pkgSourcedFromFile"
+    , poetryLockPackageVersion = "1.1.0"
+    , poetryLockPackageCategory = "main"
+    , poetryLockPackageOptional = False
+    , poetryLockPackageDependencies = Map.empty
+    , poetryLockPackagePythonVersions = "*"
+    , poetryLockPackageSource =
+        Just
+          ( PoetryLockPackageSource
+              "file"
+              "pkgFile-1.1.0.tar.gz"
+              Nothing
+              Nothing
+          )
+    }
+
+expectedDeps :: [Dependency]
+expectedDeps =
+  [ dep PipType "flake8" (Just $ CCompatible "1.1") prodEnvs
+  , dep GitType "https://github.com/pallets/flask.git" (Just $ CEq "38eb5d3b") prodEnvs
+  , dep GitType "https://github.com/networkx/networkx.git" Nothing prodEnvs
+  , dep GitType "https://github.com/numpy/numpy.git" (Just $ CEq "v0.13.2") prodEnvs
+  , dep GitType "https://github.com/kennethreitz/requests.git" (Just $ CEq "next") prodEnvs
+  , dep URLType "https://example.com/my-package-0.1.0.tar.gz" Nothing prodEnvs
+  , dep PipType "black" (Just $ CEq "19.10b0") prodEnvs
+  , dep PipType "pytest" Nothing devEnvs
+  ]
+  where
+    dep :: DepType -> Text -> Maybe VerConstraint -> [DepEnvironment] -> Dependency
+    dep t n v e = Dependency t n v [] e Map.empty
+    prodEnvs :: [DepEnvironment]
+    prodEnvs = [EnvProduction]
+    devEnvs :: [DepEnvironment]
+    devEnvs = [EnvDevelopment]
+
+spec :: Spec
+spec = do
+  nominalContents <- runIO (TIO.readFile "test/Python/Poetry/testdata/pyproject1.toml")
+  emptyContents <- runIO (TIO.readFile "test/Python/Poetry/testdata/pyproject2.toml")
+
+  describe "getDependencies" $ do
```

Comment:
[nit]: Lot's of unneeded `do`'s here.  If it compiles without it, it's safe to remove.